backend/src/bundles/avatar-videos/avatar-videos.controller.ts

```diff
@@ -0,0 +1,134 @@
+import { type UserGetCurrentResponseDto, ApiPath } from 'shared';
+
+import {
+    type ApiHandlerOptions,
+    type ApiHandlerResponse,
+} from '~/common/controller/controller.js';
+import { BaseController } from '~/common/controller/controller.js';
+import { HttpCode, HTTPMethod } from '~/common/http/http.js';
+import { type Logger } from '~/common/logger/logger.js';
+
+import { type AvatarVideoService } from './avatar-videos.service.js';
+import { AvatarVideosApiPath } from './enums/enums.js';
+import {
+    type GetAvatarVideoRequestDto,
+    type RenderAvatarVideoRequestDto,
+} from './types/types.js';
+import { renderAvatarVideoValidationSchema } from './validation-schemas/validation-schemas.js';
+
+class AvatarVideoController extends BaseController {
+    private avatarVideoService: AvatarVideoService;
+
+    public constructor(logger: Logger, avatarVideoService: AvatarVideoService) {
+        super(logger, ApiPath.AVATAR_VIDEO);
+
+        this.avatarVideoService = avatarVideoService;
+
+        this.addRoute({
+            path: AvatarVideosApiPath.ID,
+            method: HTTPMethod.GET,
+            handler: (options) =>
+                this.getAvatarVideo(
+                    options as ApiHandlerOptions<{
+                        params: GetAvatarVideoRequestDto;
+                    }>,
+                ),
+        });
+
+        this.addRoute({
+            path: AvatarVideosApiPath.ROOT,
+            method: HTTPMethod.POST,
+            validation: {
+                body: renderAvatarVideoValidationSchema,
+            },
+            handler: (options) =>
+                this.renderAvatarVideo(
+                    options as ApiHandlerOptions<{
+                        body: RenderAvatarVideoRequestDto;
+                    }>,
+                ),
+        });
+    }
+
+    /**
+     * @swagger
+     * /avatar-video/:id:
+     *    get:
+     *      description: Get video url
+     *      security:
+     *       - bearerAuth: []
+     *      responses:
+     *        200:
+     *          description: Successful operation
+     *          content:
+     *            application/json:
+     *              schema:
+     *                type: object
+     *                properties:
+     *                  url:
+     *                    type: string
+     */
+    private async getAvatarVideo(
+        options: ApiHandlerOptions<{
+            params: GetAvatarVideoRequestDto;
+        }>,
+    ): Promise<ApiHandlerResponse> {
+        return {
+            payload: await this.avatarVideoService.getAvatarVideo({
+                id: options.params.id,
+            }),
+            status: HttpCode.OK,
+        };
+    }
```

Comment:
Do we need a separate controller to get avatar video by id? I thought once it is generated we save it to videos table and then on home page just retrieve all videos and show them to user. And on frontend we will now anything about generated video id.